src/cascadia/PublicTerminalCore/Source.cpp

```diff
@@ -0,0 +1,291 @@
+#include "HwndTerminal.hpp"
+#include <DefaultSettings.h>
+#include "../../renderer/base/Renderer.hpp"
+#include "../../renderer/dx/DxRenderer.hpp"
+#include "../../cascadia/TerminalCore/Terminal.hpp"
+#include "../../cascadia/inc/cppwinrt_utils.h"
+#include "../../types/viewport.cpp"
+#include "..\..\types\inc\GlyphWidth.hpp"
+
+using namespace ::Microsoft::Terminal::Core;
+
+static WCHAR szTOOLBOXWNDCLASS[] = L"HwndTerminalClass";
+
+LRESULT CALLBACK HwndTerminalWndProc(
+    HWND hwnd,
+    UINT uMsg,
+    WPARAM wParam,
+    LPARAM lParam)
+{
+    //HwndTerminal* terminal = (HwndTerminal*)GetWindowLongPtr(hwnd, GWLP_USERDATA);
+    //if (terminal)
+    //{
+    //    switch (uMsg)
+    //    {
+    //    case WM_MOUSEACTIVATE:
+    //        SetFocus(terminal->_hwnd);
+    //        return 0;
+    //    }
+    //}
+
+    return DefWindowProc(hwnd, uMsg, wParam, lParam);
+}
+
+HwndTerminal::HwndTerminal(HWND parentHwnd) :
+    _desiredFont{ DEFAULT_FONT_FACE.c_str(), 0, 10, { 0, 14 }, CP_UTF8 },
+    _actualFont{ DEFAULT_FONT_FACE.c_str(), 0, 10, { 0, 14 }, CP_UTF8, false }
+{
+    WNDCLASSW wc;
+
+    wc.style = 0;
+    wc.lpfnWndProc = HwndTerminalWndProc;
+    wc.cbClsExtra = 0;
+    wc.cbWndExtra = 0;
+    wc.hInstance = GetModuleHandle(nullptr);
+    wc.hIcon = nullptr;
+    wc.hCursor = LoadCursor(nullptr, IDC_ARROW);
+    wc.hbrBackground = nullptr;
+    wc.lpszMenuName = nullptr;
+    wc.lpszClassName = szTOOLBOXWNDCLASS;
+
+    bool registered = RegisterClassW(&wc) != 0;
+
+    if (registered)
+    {
+        auto h = CreateWindowExW(
+            0,
+            szTOOLBOXWNDCLASS,
+            nullptr,
+            WS_CHILD |
+                WS_CLIPCHILDREN |
+                WS_CLIPSIBLINGS |
+                WS_VISIBLE,
+            0,
+            0,
+            50,
+            50,
+            parentHwnd,
+            nullptr,
+            GetModuleHandle(nullptr),
+            nullptr);
+
+        _hwnd = h;
+
+        SetWindowLongPtr(_hwnd, GWLP_USERDATA, (LONG_PTR)this);
+    }
+}
+
+HwndTerminal::~HwndTerminal()
+{
+    DestroyWindow(_hwnd);
+}
+
+HRESULT HwndTerminal::Initialize()
+{
+    _terminal = std::make_unique<::Microsoft::Terminal::Core::Terminal>();
+    auto renderThread = std::make_unique<::Microsoft::Console::Render::RenderThread>();
+    auto* const localPointerToThread = renderThread.get();
+    _renderer = std::make_unique<::Microsoft::Console::Render::Renderer>(_terminal.get(), nullptr, 0, std::move(renderThread));
+
+    RETURN_IF_FAILED(localPointerToThread->Initialize(_renderer.get()));
+
+    auto dxEngine = std::make_unique<::Microsoft::Console::Render::DxEngine>();
+    RETURN_IF_FAILED(dxEngine->SetHwnd(_hwnd));
+    RETURN_IF_FAILED(dxEngine->Enable());
+    _renderer->AddRenderEngine(dxEngine.get());
+
+    auto pfn = std::bind(&::Microsoft::Console::Render::Renderer::IsGlyphWideByFont, _renderer.get(), std::placeholders::_1);
+    SetGlyphWidthFallback(pfn);
+
+    _UpdateFont(96);

```

Comment:
96 is a magic value. use the `USER_DEFAULT_SCREEN_DPI` constant if this is supposed to be 96dpi.